src/components/_tests/ContextBar.test.js

```diff
@@ -0,0 +1,561 @@
+  googleProject: 'terra-dev-98897219',
+  cloudContext: {
+    cloudProvider: 'GCP',
+    cloudResource: 'terra-dev-98897219'
+  },
+  auditInfo: {
+    creator: 'ncl.hedwig@gmail.com',
+    createdDate: '2022-09-08T19:46:37.396597Z',
+    destroyedDate: null,
+    dateAccessed: '2022-09-08T19:47:21.206Z'
+  },
+  runtimeConfig: {
+    machineType: 'n1-standard-4',
+    persistentDiskId: 15774,
+    cloudService: 'GCE',
+    bootDiskSize: 120,
+    zone: 'us-central1-a',
+    gpuConfig: null
+  },
+  proxyUrl: 'https://leonardo.dsde-dev.broadinstitute.org/proxy/terra-dev-98897219/saturn-48afb74a-15b1-4aad-8b23-d039cf8253fb/rstudio',
+  status: 'Creating',
+  labels: {
+    saturnWorkspaceNamespace: 'general-dev-billing-account',
+    'saturn-iframe-extension': 'https://bvdp-saturn-dev.appspot.com/jupyter-iframe-extension.js',
+    creator: 'ncl.hedwig@gmail.com',
+    clusterServiceAccount: 'pet-26534176105071279add1@terra-dev-98897219.iam.gserviceaccount.com',
+    saturnAutoCreated: 'true',
+    clusterName: 'saturn-48afb74a-15b1-4aad-8b23-d039cf8253fb',
+    saturnWorkspaceName: 'N8s Space',
+    saturnVersion: '6',
+    tool: 'RStudio',
+    runtimeName: 'saturn-48afb74a-15b1-4aad-8b23-d039cf8253fb',
+    cloudContext: 'Gcp/terra-dev-98897219',
+    googleProject: 'terra-dev-98897219'
+  },
+  patchInProgress: false
+}
+
+const contextBarProps = {
+  runtimes: [],
+  apps: [],
+  appDataDisks: [],
+  refreshRuntimes: () => '',
+  location: 'US-CENTRAL1',
+  locationType: '',
+  refreshApps: () => '',
+  workspace: {
+    workspace: {
+      namespace: 'namespace'
+    },
+    namespace: 'Broad Test Workspace'
+  }
+}
+
+describe('ContextBar - buttons', () => {
+  it('will render default icons', () => {
+    // Act
+    render(h(ContextBar, contextBarProps))
+
+    // Assert
+    expect(screen.getByText('Rate:'))
+    expect(screen.getByLabelText('Environment Configuration'))
+    expect(screen.getByLabelText('Terminal button')).toHaveAttribute('disabled')
+  })
+
+  it('will render Jupyter button with an enabled Terminal Button', () => {
+    // Arrange
+    const jupyterContextBarProps = {
+      ...contextBarProps,
+      runtimes: [jupyter1],
+      persistentDisks: [jupyter1Disk]
+    }
+
+    // Act
+    render(h(ContextBar, jupyterContextBarProps))
+
+    //Assert
+    expect(screen.getByText('Rate:'))
+    expect(screen.getByLabelText('Environment Configuration'))
+    expect(screen.getByLabelText(new RegExp(/Jupyter Environment/i)))
+    expect(screen.getByLabelText('Terminal button')).toBeEnabled()
+  })
+
+  it('will render Galaxy and RStudio buttons with a disabled Terminal Button', () => {
+    // Arrange
+    const rstudioGalaxyContextBarProps = {
+      ...contextBarProps,
+      runtimes: [rstudioRuntime],
+      apps: [galaxyRunning],
+      appDataDisks: [galaxyDisk],
+      persistentDisks: [jupyter1Disk]
+    }
+
+    // Act
+    render(h(ContextBar, rstudioGalaxyContextBarProps))
```

Comment:
do you not need to do `await act(async () => render(...))` here?

Reply:
I haven't had a problem with it in it's current state